Add cat command

tardypad opened the pull request:
There are some types of notes which are meant to be viewed most of time and edited only rarely
It's inconvenient to have to open an editor just to see them so I've added this show command to display the content of a note in the standard output

I believe everything needed is there: code, completion, test, doc
I have made to play nicely with pipes like the open command

pimterry commented on test/test-show.bats:
```diff
@@ -0,0 +1,81 @@
+}
+
+@test "Should fail to show non-existent note" {
+  run $notes rm note
```
this should be `show`

pimterry commented on test/test-show.bats:
```diff
@@ -0,0 +1,81 @@
+  assert_output $'line1\nline2'
+}
+
+@test "Should complain and ask for a name if not is provided to show" {
```
typo, should be `...if one is not provided to show`

pimterry commented on notes:
```diff
@@ -167,6 +167,33 @@ open_note() {
+    fi
+    if [ ! -f "$note_path" ]; then
+        note_path="$notes_dir/$note_path"
+    fi
```
Both for both of these functions, most of the logic is basically the same as the `open_` equivalent. Can we commonize these, rather the duplicating the `$note_path` and `/dev/stdin` handing?

tardypad commented:
I've renamed the command to `cat`
I thought of it initially but then some command were not following this schema too like `search` and `open` so I went for `show` (which sounds more explicit than `cat`) but actually I just realized that there is no dedicated commands linked to those previous ones.
So overall, naming this one `cat` is indeed the most consistent choice.

I've fixed the typos you mentioned and grouped the code as well.
I thought it would be more complex than this, that's why I didn't do it in the first place, but in the end it looks fine.

pimterry commented on test/test-cat.bats:
```diff
@@ -0,0 +1,81 @@
+}
+
+@test "Accepts relative notes paths to show" {
+  touch $NOTES_DIRECTORY/note.md
```
Here (and I think everywhere in this file?) you don't need the `touch`, since you're echoing straight afterwards it's not actually doing anything.

tardypad replied:
Indeed, good point.
Fixed now
